src/compute/src/render/reduce.rs

```diff
@@ -702,12 +734,43 @@ where
                             );
                         }
                     }
+                    // After complaining, output an error here so that we can eventually
+                    // report it in an error stream.
+                    target.push((Err(format!("Invalid data in source, saw negative accumulation for key {key:?} in hierarchical mins-maxes aggregate")), 1));
                 } else {
                     let mut output = Vec::with_capacity(aggrs.len());
                     for (aggr_index, func) in aggrs.iter().enumerate() {
                         let iter = source.iter().map(|(values, _cnt)| values[aggr_index].iter().next().unwrap());
                         output.push(Row::pack_slice(&[func.eval(iter, &RowArena::new())]));
                     }
+                    // We do exactly the same here as in levels of the hierarchical reduction
+                    // in which we do not output `Result` (see below), except that correct data
+                    // output needs to be surrounded by `Ok` and multiplicities reported normally.
+                    target.push((Ok(output), -1));
+                    target.extend(source.iter().map(|(values, cnt)| (Ok((*values).clone()), *cnt)));
+                }
+            }
+        })
+        .as_collection(|k,v| (k.clone(), v.clone()))
+        .map_fallible("Checked Invalid Accumulations", |(key, result)| {
+            use mz_expr::EvalError;
+            match result {
+                Err(message) => Err(DataflowError::EvalError(Box::new(EvalError::Internal(message)))),
```

Comment:
```suggestion
                Err(message) => Err(DataflowError::from(EvalError::Internal(message))),
```
... or:
```suggestion
                Err(message) => Err(EvalError::Internal(message).into()),
```

Reply:
Thanks! I went with the second suggestion.